### PR TITLE
Add FpType to CFA readme

### PR DESCRIPTION
**Pull request by leventeBajczi**

### CODE REVIEW

File: subprojects/cfa/cfa/README.md [hunk 1]
```diff
@@ -39,18 +39,31 @@ Variables of the CFA can have the following types.
 * `rat`: Rational numbers (implemented as SMT reals).
 * `[K] -> V`: SMT arrays (associative maps) from a given key type `K` to a value type `V`.
 * `bv[L]`: Bitvector of given length `L`. _This is an experimental feature. See the [details](doc/bitvectors.md) for more information._
+* `fp[E:S]`: Floating point of given size exponent `E` and significand `S`. Significand size should include the hidden bit as well.`
```
Comment (hajduakos) [inline subprojects/cfa/cfa/README.md:42]: There is an extra backtick at the end of the line

Comment (hajduakos) [inline subprojects/cfa/cfa/README.md:42]: Can you maybe put a reference here to any standards that this type follows? And it is also worth mentioning that this type corresponds to the SMT LIB theory of floats.

File: subprojects/cfa/cfa/README.md [hunk 2]
```diff
@@ -39,18 +39,31 @@ Variables of the CFA can have the following types.
...
+  * Floating point literals can be given in the following form: `[+-]?bitvector.bitvector`, defining the optional sign (+/-), exponent and significand in this order. (E.g. `+5'b10000.10'd0` is `+2.0f` of type `fp[5:11]` (half-precision IEEE-754 float)). The value is calculated the following way (using `e` for exponent size, and `E`, `S` for values, `h` for sign): `value = (-1)^h * 1.S * 2^(E-2^(e-1)+1)`
 * Comparison, e.g., `=`, `/=`, `<`, `>`, `<=`, `>=`.
 * Boolean operators, e.g., `and`, `or`, `xor`, `not`, `imply`, `iff`.
 * Arithmetic, e.g., `+`, `-`, `/`, `*`, `mod`, `rem`.
 * Conditional: `if . then . else .`.
 * Array read (`a[i]`) and write (`a[i <- v]`).
 * Bitvector specific operators, e.g., `bvand`, `bvor`, `bvxor`, `bvshl`, `bvashr`, `bvlshr`, `bvrol`, `bvror`, `bvnot`, etc. _This is an experimental feature. See the [details](doc/bitvectors.md) for more information._
+* Floating point operators are special in the sense that extra arguments can be passed to them via brackets.
```
Comment (hajduakos) [inline subprojects/cfa/cfa/README.md:56]: I would group these operators based on their kind (e.g., arithmetic, comparisons, rounding, conversions, etc.) and then have the explanation on arguments afterwards (that is already there).

**Review by hajduakos — CHANGES_REQUESTED** [review]
Looks good, thanks! Just requested some small changes/additions.

**Comment by leventeBajczi** [comment]
I've applied the suggestions above, could you please check it again?

**Review by hajduakos — APPROVED** [review]
Thanks for the changes! Looks good.

**Pull request merged**